packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx

```diff
@@ -1,427 +1,187 @@
-              <div className={ css('ms-CommandBarItem', styles.item) } key={ OVERFLOW_KEY } ref={ OVERFLOW_KEY }>
-                <button
-                  type='button'
-                  id={ this._id + OVERFLOW_KEY }
-                  className={ css('ms-CommandBarItem-link', styles.itemLink, {
-                    ['is-expanded ' + styles.itemLinkIsExpanded]: (expandedMenuItemKey === OVERFLOW_KEY)
-                  }) }
-                  onClick={ this._onOverflowClick }
-                  role='menuitem'
-                  aria-label={ this.props.elipisisAriaLabel || '' }
-                  aria-haspopup={ true }
-                  data-automation-id='commandBarOverflow'
-                >
-                  <Icon className={ css('ms-CommandBarItem-overflow', styles.itemOverflow) } iconName='more' />
-                </button>
-              </div>
-            ] : []) }
-          </div>
-          <div className={ css('ms-CommandBar-sideCommands', styles.sideCommands) } ref='farCommandSurface'>
-            { renderedFarItems!.map((item, index) => (
-              this._renderItemInCommandBar(item, index, expandedMenuItemKey!, true)
-            )) }
-          </div>
-        </FocusZone>
-        { (contextualMenuProps) ?
-          (<ContextualMenu
-            className={ css('ms-CommandBar-menuHost') }
-            directionalHint={ DirectionalHint.bottomAutoEdge }
-            { ...contextualMenuProps }
-            targetElement={ contextualMenuTarget }
-            labelElementId={ expandedMenuId }
-            onDismiss={ this._onContextMenuDismiss }
-          />
-          ) : (null)
-        }
-      </div >
+      <ResizeGroup
+        data={ commandBardata }
+        onReduceData={ onReduceData }
+        onGrowData={ onGrowData }
+        onRenderData={ (data: ICommandBarData) => {
+          return (
+            <div className={ css('ms-CommandBar', styles.root) }>
+
+              {/*Primary Items*/ }
+              <OverflowSet
+                ref='overflowSet'
+                className={ css(styles.primarySet) }
+                items={ data.primaryItems }
+                overflowItems={ data.overflowItems.length ? data.overflowItems : undefined }
+                onRenderItem={ this._onRenderItems }
+                onRenderOverflowButton={ (renderedOverflowItems: ICommandBarItemProps[]) => {
+                  return (
+                    this._onRenderButton({
+                      key: 'oveflowButton',
+                      styles: { buttonStyles },
+                      ariaLabel: elipisisAriaLabel,
+                      className: css(styles.overflowButton),
+                      menuProps: { items: renderedOverflowItems },
+                      menuIconProps: elipisisIconProps,
+                    })
+                  );
+                } }
+              />
+
+              {/*Secondary Items*/ }
+              <OverflowSet
```

Comment:
So the far commands never render in an overflow. I'm wondering if this makes sense to be taken out of command bar and leaving command bar to just deal with the part that resizes? Consumers could still implement a far commands by composing multiple components